pkg/kube_events_manager/filter.go

```diff
@@ -31,18 +31,24 @@ func ApplyFilter(jqFilter string, filterFn func(obj *unstructured.Unstructured)
 	if err != nil {
 		return nil, err
 	}
-	res.ObjectBytes = int64(len(data))
+	res.ObjectSize = len(data)
 
 	// If filterFn is passed, run it and return result.
 	if filterFn != nil {
-		var err error
-		var filtered string
-		filtered, err = filterFn(obj)
+		filteredObj, err := filterFn(obj)
 		if err != nil {
 			return nil, fmt.Errorf("filterFn: %v", err)
 		}
-		res.FilterResult = filtered
-		res.Metadata.Checksum = utils_checksum.CalculateChecksum(filtered)
+
+		filteredBytes, err := json.Marshal(filteredObj)
+		if err != nil {
+			return nil, err
+		}
+
+		res.FilterResult = filteredObj
```

Comment:
Just curious. Do we have a test for filter func and its memory measurement?

Reply:
There isn't a test for jqFilter or filterFunc.